server/src/controllers/roomsController.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import express from 'express';
+import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { CommunicationAccessToken, CommunicationIdentityClient, TokenScope } from '@azure/communication-identity';
+import { Room, RoomsClient, RoomParticipant, CreateRoomOptions } from '@azure/communication-rooms';
+import {
+  CreateTestAppointmentResponse,
+  JoinTestAppointmentResponse,
+  RoomParticipantRole,
+  RoomParticipant as TestAppointmentRoomParticipant
+} from '../models/roomModel';
+import { ERROR_NO_USER_FOUND_IN_ROOM } from '../constants';
+
+export const createRoom = (identityClient: CommunicationIdentityClient, roomsClient: RoomsClient) => async (
+  _req: express.Request,
+  res: express.Response,
+  next: express.NextFunction
+): Promise<any> => {
+  try {
+    const presenter = await identityClient.createUser();
+    const attendee = await identityClient.createUser();
+
+    // Options payload to create a room
+    const createRoomOptions: CreateRoomOptions = {
+      participants: [
+        {
+          id: presenter,
+          role: RoomParticipantRole.presenter
+        },
+        {
+          id: attendee,
+          role: RoomParticipantRole.attendee
+        }
+      ]
+    };
+
+    // Create a room with the request payload
+    const createdRoom: Room = await roomsClient.createRoom(createRoomOptions);
+
+    // Formulating participants
+    const participants = createdRoom.participants.map(
+      (participant: RoomParticipant): TestAppointmentRoomParticipant => ({
+        id: (participant.id as CommunicationUserIdentifier).communicationUserId as string,
+        role: participant.role as RoomParticipantRole
+      })
+    );
+
+    // Formulate response
+    const response: CreateTestAppointmentResponse = {
+      roomId: createdRoom.id,
+      participants: participants
+    };
+
+    return res.send(response);
+  } catch (error) {
+    return next(error);
+  }
+};
+
+export const joinRoom = (identityClient: CommunicationIdentityClient, roomsClient: RoomsClient) => async (
+  req: express.Request,
+  res: express.Response,
+  next: express.NextFunction
+): Promise<any> => {
+  try {
+    const { roomId, userId } = req.query;
+
+    // Retrieve participants list
+    const participantsList = await roomsClient.getParticipants(roomId as string);
+
+    // Check if the user is part of participants
+    const foundUserParticipant: RoomParticipant | undefined = participantsList.find(
+      (participant: RoomParticipant) => (participant.id as CommunicationUserIdentifier).communicationUserId === userId
+    );
+
+    if (foundUserParticipant) {
+      // Create token
+      const scopes: TokenScope[] = ['voip'];
```

Comment:
Do we need chat scope here as well?

Reply:
I am afraid that this is necessary.

Reply:
I will double check it tomorrow!

Reply:
It looks like the `TokenScope` is required parameter for `getToken` as below:
![image](https://user-images.githubusercontent.com/94404633/199550134-0640e6b3-398c-413a-8fc3-33bae30cc545.png)

Reply:
I mean like this `const scopes: TokenScope[] = ['voip', 'chat'];`

Reply:
I don't think Rooms supports chat. Also, I don't see the `chat` option in the sample and [documentation](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/access-tokens?tabs=windows&pivots=programming-language-javascript#issue-access-tokens).

![image](https://user-images.githubusercontent.com/94404633/199791557-088bdbee-4e27-4119-8e1f-887e0d222618.png)